cpp/src/traversal/extract_bfs_paths_impl.cuh

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <utilities/graph_utils.cuh>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/detail/graph_utils.cuh>
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/utilities/error.hpp>
+#include <cugraph/utilities/host_scalar_comm.cuh>
+#include <cugraph/utilities/shuffle_comm.cuh>
+#include <cugraph/vertex_partition_device_view.cuh>
+
+#include <raft/handle.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/binary_search.h>
+#include <thrust/fill.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/transform.h>
+#include <thrust/tuple.h>
+
+namespace cugraph {
+
+namespace detail {
+
+template <typename vertex_t>
+struct mg_partition_vertices {
+  int rank_;
+  vertex_t const* d_partition_offsets_begin_;
+  vertex_t const* d_partition_offsets_end_;
+
+  int __device__ operator()(vertex_t v) const
+  {
+#if 0
```

Comment:
This needs either a FIXME to drop/cleanup or should be a DEBUG or other defined type rather than "0"

Reply:
Already pushed an update to remove this.